relations/src/r1cs/constraint_system.rs

```diff
@@ -356,11 +359,11 @@ impl<F: Field> ConstraintSystem<F> {
                     self.witness_assignment.push(acc);
                 }
 
-                self.num_constraints += 1;
-
-                additional_a_constraints.push(index);
-                additional_c_constraints_witness_assignments.push(witness_assignment);
+                // Add a new constarint for this new witness
+                new_witness_linear_combinations.push(outlined_lc.clone());
+                new_witness_indices.push(witness_assignment);
```

Comment:
```suggestion
                // Add a new constraint for this new witness
                new_witness_linear_combinations.push(outlined_lc.clone());
                new_witness_indices.push(witness_assignment);
```

Also, why not use `self.constrain` here?